libdrgn/program.c

```diff
@@ -513,6 +514,54 @@ drgn_program_check_initialized(struct drgn_program *prog)
 	return NULL;
 }
 
+static struct drgn_error *
+drgn_program_set_kdump(struct drgn_program *prog)
+{
+	kdump_ctx_t *ctx = NULL;
+	struct drgn_error *err = drgn_kdump_init(&ctx, prog->core_fd);
+	if (err)
+		goto out_fd;
+
+	err = drgn_kdump_get_kernel_offset(ctx,
+	                                   &prog->vmcoreinfo.kaslr_offset);
+	if (err)
+		goto out_kdump;
+
+	err = drgn_kdump_get_osrelease(ctx, prog->vmcoreinfo.osrelease);
+	if (err)
+		goto out_kdump;
+
+	err = drgn_kdump_get_page_size(ctx, &prog->vmcoreinfo.page_size);
+	if (err)
+		goto out_kdump;
```

Comment:
This is the only part I don't like at a high level. drgn already has code for parsing vmcoreinfo, so I'd rather use that and avoid having to add a new call here for every field we start using. We'll probably just need to do some minimal parsing straight from the fd.

Reply:
Great idea. Decided to use `kdump_vmcoreinfo_raw()` provided by `libkdumpfile` to get the whole thing as a string and then used `parse_vmcoreinfo()` which is provided by `drgn`. Does this work better?